packages/drivers/odsp-driver/src/odspDocumentStorageServiceBase.ts

```diff
@@ -189,6 +192,10 @@ export abstract class OdspDocumentStorageServiceBase implements IDocumentStorage
 		}
 
 		const snapshotTree = await this.readTree(id, scenarioName);
+		// Remove it once it is read.
+		if (isPartialSnapshot(snapshotTree)) {
+			this.commitCache.delete(id);
```

Comment:
shouldn't it get into the cache in the first place?

Reply:
BTW, ideally it should be not a function of type of snapshot. It should a function of what API is called on the driver. Old code and new code would need to call different API (or supply different / extra args) to clearly describe intention (yes, caller can understand partial snapshot and does not need caching).

The rest of the code can clearly assume that partial snapshots could show up in workflow only if client called that new API, and thus do checks like that, but that's not ideal. I'd rather be more explicit and pass that kind of info as arguments around (can / should it be cached) and not depend on type of snapshot. That allows us (for example) to start using new types even if everything else works exactly the same (with assumption that snapshots are still full)